### ISSUE TITLE
No support for Stress data

### ISSUE THREAD

**Issue by MakisH** [precice/calculix-adapter#111]
Our [Turek Hron FSI3 tutorial](https://precice.org/tutorials-turek-hron-fsi3.html) (OpenFOAM-deal.II) exchanges stress and displacement data. However, CalculiX does not currently know how to handle stress data. Supporting stresses would make adding a CalculiX case to this tutorial much easier.

**Comment by boris-martin** [comment]
Do you want to make the adapter able to *write* stress ? Because reading Pressure is already possible since https://github.com/precice/calculix-adapter/pull/91
But AFAIK solids write displacements and read stresses/pressure, so this should be fine

**Comment by MakisH** [comment]
Oh, then we should probably add the word "stress" somewhere in this page: https://precice.org/adapter-calculix-config.html

We only need to read stress. I assume that pressure is only a scalar with the normal component. Is that correct?

**Comment by boris-martin** [comment]
Yes. So no tangential traction is possible here. I don't know if it's a CalculiX limitation or something else.
Maybe it's useful to have a wrapper that reads a vector input and takes the normal part ?
(But it's actually "reading the full stress tensor" I think... but preCICE doesn't support tensor data)

**Comment by MakisH** [comment]
> Yes. So no tangential traction is possible here. I don't know if it's a CalculiX limitation or something else.

I don't know if CalculiX has any such limitation. Maybe @MatthiasFreimuth has some clue here.

> Maybe it's useful to have a wrapper that reads a vector input and takes the normal part ?

What do you mean by "wrapper"? This already sounds doable with [preCICE actions](https://precice.org/configuration-action.html).

> (But it's actually "reading the full stress tensor" I think... but preCICE doesn't support tensor data)

But then how do we do it in deal.II? @davidscn

**Comment by boris-martin** [comment]
> > Maybe it's useful to have a wrapper that reads a vector input and takes the normal part ?
> 
> What do you mean by "wrapper"? This already sounds doable with [preCICE actions](https://precice.org/configuration-action.html).

Didn't think of actions, I guess it could work!
By wrapper I meant updating the adapter to add a conversion layer, but actions are probably cleaner

**Comment by MatthiasFreimuth** [comment]
As far as I know, there should be no limitations from CalculiX side regarding tangential traction.

**Comment by davidscn** [comment]
> But then how do we do it in deal.II? @davidscn

We just read a Stress vector. On the OpenFOAM side we also compute it via normal vector x stress tensor.
